.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,40 @@
+# This workflow will do a clean install of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+name: Node.js CI - Build + Test
+
+on:
+  push:
+    branches:
+      - "*"
+  pull_request:
+    branches:
+      - "*"
```

Comment:
This will trigger the action twice for any new pull request. I would suggest to put on pull request to main/develop branch instead

Or on pull request to any branch and on push on specific branch (develop)